### ISSUE TITLE
Add a capability checking system and command unregistration

### ISSUE THREAD

**Issue by simpleauthority** [Incendo/cloud#175]
Command unregistration is a sorely needed feature. My use case is a simple text-command remembrance system that registers simple alias->output commands. There is another command which "forgets" these aliased commands, and I need to unregister in cloud in that instance.

As I understand that MC platforms do not typically react well to unregistration (brigadier) I suggest a capability system.

From @Proximyst, it would look like this:
```
commandManager.capability(CommandUnregistration.class).ifPresent(cap -> cap.unregister(command))
```

This seems perfectly reasonable. Please add. Thanks =))

**Comment by bergerkiller** [comment]
I personally need this for modular components. Others can extend my plugin's API, possibly with custom cloud-based commands. This requires proper support for plugins, and also methods of un-registering handlers when the plugin disables itself.

You can also argue that it should be possible to turn such modules on or off at runtime, using configuration. For example, an extension could be toggled on or off through configuration, at which a subsection of commands become available or disappear.

In addition to being able to unregister commands, it would be really helpful if I could store commands in a temporary immutable standalone object. That way I can track what commands are registered to what module, and instantly unregister all of them.

**Comment by Citymonstret** [comment]
This has been implemented in 1.7.0. 

```java
if (this.manager.hasCapability(CloudCapability.StandardCapabilities.ROOT_COMMAND_DELETION)) {
    this.manager.deleteRootCommand("thecommand");
}
```
